src/deezer/client.py

```diff
@@ -118,12 +118,11 @@ def _process_json(
         if "id" not in result and resource_id is not None:
             result["id"] = resource_id
 
-        if "type" in result:
-            if result["type"] in self.objects_types:
-                object_class = self.objects_types[result["type"]]
-            else:
-                # in case any new types are introduced by the API
-                object_class = Resource
+        if "type" in result and result["type"] in self.objects_types:
+            object_class = self.objects_types[result["type"]]
+        elif "type" in result or not resource_type and "id" in result:
+            # in case any new types are introduced by the API
+            object_class = Resource
```

Comment:
Note to myself: not sure I agree with Sourcery here...😕 

(Don't worry about changing it back, it's fine)

Reply:
Yeah I also thought it looked weird